### PR TITLE
[now dev] Add support for `handle: miss` and `handle: hit`

### PR DESCRIPTION
**Pull request by styfle**
- [x] Add tests from now-proxy for `handle: miss`
- [x] Add tests from now-proxy for `handle: hit`
- [x] Add file output renaming when `featHandleMiss` is true (also assign true for now dev)

### CODE REVIEW

File: packages/now-cli/test/dev/integration.js [hunk 1]
```diff
@@ -319,6 +319,53 @@ test(
...
+    t.is(override, 'one', 'exected override header to not override');
+
+    const body = await response.text();
+    t.regex(body, /Blog/gm);
+  })
+);
+
+test(
+  '[now dev] handles miss after rewrite',
+  testFixtureStdio('handle-miss-after-rewrite', async (t, port) => {
```
Comment (dav-is) [inline packages/now-cli/test/dev/integration.js:356]: Let's add some more tests for this fixture:
`/about.html` returns 200 without any added headers
`/blog/post/one.html` returns 200 without any added headers
`/blog/post` returns 200 with headers: `test: 1` and `override: 2`

Comment (styfle) [reply packages/now-cli/test/dev/integration.js:356]: Thanks, done 👍

File: packages/now-cli/src/util/dev/types.ts [hunk 2]
```diff
@@ -156,6 +157,8 @@ export interface RouteResult {
   matched_route_idx?: number;
   // "userDest": <boolean in case the destination was user defined>
   userDest?: boolean;
+  //url as destinatino should end routing
```
Comment (AndyBitz) [inline packages/now-cli/src/util/dev/types.ts]:
```suggestion
  // url as destination should end routing
```

**Review by AndyBitz — APPROVED** [review]

File: packages/now-cli/src/util/dev/router.ts [hunk 3]
```diff
@@ -74,41 +89,73 @@ export default async function(
...
 
         if (headers) {
-          // Create a clone of the `headers` object to not mutate the original one
-          headers = { ...headers };
           for (const key of Object.keys(headers)) {
-            headers[key] = resolveRouteParameters(headers[key], match, keys);
+            if (
+              previousHeaders &&
+              // eslint-disable-next-line no-prototype-builtins
+              previousHeaders.hasOwnProperty(key) &&
```
Comment (dav-is) [inline packages/now-cli/src/util/dev/router.ts]:
```suggestion
              Object.prototype.hasOwnProperty.call(previousHeaders, key) &&
```

File: packages/now-cli/src/util/dev/router.ts [hunk 4]
```diff
@@ -74,41 +89,73 @@ export default async function(
         }
 
         if (headers) {
-          // Create a clone of the `headers` object to not mutate the original one
-          headers = { ...headers };
           for (const key of Object.keys(headers)) {
-            headers[key] = resolveRouteParameters(headers[key], match, keys);
+            if (
+              previousHeaders &&
+              // eslint-disable-next-line no-prototype-builtins
```
Comment (dav-is) [inline packages/now-cli/src/util/dev/router.ts]:
```suggestion
```

**Review by dav-is — APPROVED** [review]

**Pull request merged**